vncviewer/DesktopWindow.cxx

```diff
@@ -875,50 +873,62 @@ int DesktopWindow::fltkHandle(int event, Fl_Window *win)
   return ret;
 }
 
-
 void DesktopWindow::fullscreen_on()
 {
-  if (not fullScreenAllMonitors)
-    fullscreen_screens(-1, -1, -1, -1);
-  else {
-    int top, bottom, left, right;
-    int top_y, bottom_y, left_x, right_x;
+  int top, bottom, left, right;
+  top = bottom = left = right = -1;
+  bool all_monitors = !strcasecmp(fullScreenMode, "all");
 
-    int sx, sy, sw, sh;
+  if (all_monitors) {
+      std::set<int> monitors;
 
-    top = bottom = left = right = 0;
+      for (int i = 0; i < Fl::screen_count(); i++) {
+        monitors.insert(i);
+      }
 
-    Fl::screen_xywh(sx, sy, sw, sh, 0);
-    top_y = sy;
-    bottom_y = sy + sh;
-    left_x = sx;
-    right_x = sx + sw;
+      int x, y, w, h;
```

Comment:
Could we keep these as `sx`, `sy`, ...? I think it makes the diff better so it's easier to see what is actually happening in this commit.